### ISSUE TITLE
Native script does not cope well with -no-remote

### ISSUE THREAD

**Issue by lorenzleutgeb** [tridactyl/tridactyl#1034]
Not sure if this is a problem that can be resolved at all, but `:open about:config` is not working out nicely for me, and I suspect that this is because I run Firefox with `-no-remote`.

I have two different Firefox profiles (one personal, one for work; I know about containers and use them inside profiles).

When I run just one Firefox instance and call `:open about:config`, another instance with my second profile fires up and opens `about:config`.

When there are already two instances running and I call `:open about:config` I will get an error message saying:
> Firefox is already running, but is not responding. To open a new window, you must first close the existing Firefox process, or restart your system.

Maybe you even want to consider this a non-issue, but in that case a hint about interaction with `-no-remote` in README or a wiki might be helpful.

**Comment by glacambre** [comment]
What OS do you use? If you're on Linux, could you run the following command while firefox is running with both profiles?
```
for i in $(pidof firefox) ; do echo "$(cat /proc/$i/cmdline)\n" ; done
```

**Comment by bovine3dom** [comment]
Yeah, there's no way we can fix this. We can potentially work out if you have no remote and give you an error.

The best hope is that some day Mozilla will allow us to open about things without native, as per #760.

**Comment by lorenzleutgeb** [comment]
> What OS do you use? If you're on Linux, could you run the following command while firefox is running with both profiles?
> 
> ```
> for i in $(pidof firefox) ; do echo "$(cat /proc/$i/cmdline)\n" ; done
> ```

I ran the following instead:

```
ps -a -o args | grep "^/usr/lib64/firefox/firefox" | grep -v "\-contentproc"
```

When I run the first Firefox instance myself and then execute `:open about:config`:

```
/usr/lib64/firefox/firefox -no-remote -P Personal
/usr/lib64/firefox/firefox --new-tab about:config
```

When I start both instances myself:

```
/usr/lib64/firefox/firefox -no-remote -P Personal
/usr/lib64/firefox/firefox -no-remote -P [NAME OF PROFILE REDACTED]
```

**Comment by glacambre** [comment]
Looks like there are two issues. First, `-no-remote` indeed prevents new Firefox instances from connecting to an already existing one. Since `-no-remote` isn't mandatory to run multiple profiles, you could just remove it from the command line you use to run Firefox. If you're concerned about accidentally opening a link with the wrong firefox profile, perhaps you could create a third profile and set it as default. This way if you forget the `-P profileName` argument when you're trying to run Firefox, this third profile will be used instead and you can just remove it and re-open the link with the missing `-P profileName` argument.

The second issue is with Tridactyl itself. `:open` [calls](https://github.com/tridactyl/tridactyl/blob/master/src/excmds.ts) `:nativeopen` with an empty `firefoxArgs` array. This array is later [replaced](https://github.com/tridactyl/tridactyl/blob/master/src/excmds.ts#L364-L365) with `["--new-tab"]`. Tridactyl should probably try to add `-P profileName` to it, first guessing the profile name with (ffargs)[https://github.com/tridactyl/tridactyl/blob/371f76077dae90c53c374c14c9ab440817ec5b0e/src/native_background.ts#L342] and if it doesn't work fetching it from `config.get("profiledir")` and reading `~/.mozilla/firefox/profiles.ini`. Or we could just be lazy and tell users to do something like `:set browser firefox -P profileName `, but it's a pretty terrible idea IMO.
